core/jvm/src/Buffer.kt

```diff
@@ -21,557 +21,281 @@
+  public actual fun clear(): Unit = commonClear()
 
-  @Throws(EOFException::class)
-  override fun readFully(sink: Buffer, byteCount: Long): Unit = commonReadFully(sink, byteCount)
+  public actual override fun skip(byteCount: Long): Unit = commonSkip(byteCount)
 
-  @Throws(IOException::class)
-  override fun readAll(sink: RawSink): Long = commonReadAll(sink)
+  actual override fun write(
+    source: ByteArray,
+    offset: Int,
+    byteCount: Int
+  ): Buffer = commonWrite(source, offset, byteCount)
 
-  override fun readUtf8() = readString(size, Charsets.UTF_8)
+  override fun writeAll(source: RawSource): Long = commonWriteAll(source)
 
-  @Throws(EOFException::class)
-  override fun readUtf8(byteCount: Long) = readString(byteCount, Charsets.UTF_8)
+  actual override fun write(source: RawSource, byteCount: Long): Buffer =
+    commonWrite(source, byteCount)
 
-  override fun readString(charset: Charset) = readString(size, charset)
+  actual override fun writeByte(byte: Byte): Buffer = commonWriteByte(byte)
 
-  @Throws(EOFException::class)
-  override fun readString(byteCount: Long, charset: Charset): String {
-    require(byteCount >= 0 && byteCount <= Integer.MAX_VALUE) { "byteCount: $byteCount" }
-    if (size < byteCount) throw EOFException()
-    if (byteCount == 0L) return ""
+  actual override fun writeShort(short: Short): Buffer = commonWriteShort(short)
 
-    val s = head!!
-    if (s.pos + byteCount > s.limit) {
-      // If the string spans multiple segments, delegate to readBytes().
-      return String(readByteArray(byteCount), charset)
-    }
+  actual override fun writeInt(int: Int): Buffer = commonWriteInt(int)
 
-    val result = String(s.data, s.pos, byteCount.toInt(), charset)
-    s.pos += byteCount.toInt()
-    size -= byteCount
+  actual override fun writeLong(long: Long): Buffer = commonWriteLong(long)
 
-    if (s.pos == s.limit) {
-      head = s.pop()
-      SegmentPool.recycle(s)
-    }
+  internal actual fun writableSegment(minimumCapacity: Int): Segment =
+    commonWritableSegment(minimumCapacity)
 
-    return result
-  }
+  override fun write(source: Buffer, byteCount: Long): Unit = commonWrite(source, byteCount)
 
-  @Throws(EOFException::class)
-  override fun readUtf8Line(): String? = commonReadUtf8Line()
+  override fun read(sink: Buffer, byteCount: Long): Long = commonRead(sink, byteCount)
 
-  @Throws(EOFException::class)
-  override fun readUtf8LineStrict() = readUtf8LineStrict(Long.MAX_VALUE)
+  public actual override fun flush(): Unit = Unit
 
-  @Throws(EOFException::class)
-  override fun readUtf8LineStrict(limit: Long): String = commonReadUtf8LineStrict(limit)
+  actual override fun close(): Unit = Unit
 
-  @Throws(EOFException::class)
-  override fun readUtf8CodePoint(): Int = commonReadUtf8CodePoint()
+  /**
+   * Returns a human-readable string that describes the contents of this buffer. Typically, this
+   * is a string like `[text=Hello]` or `[hex=0000ffff]`.
```

Comment:
It would be nice to know if it can be used as a unique identifier (e.g. to compare two arbitrary buffers in an ad-hoc manner)

Reply:
In Okio two UTF-8 buffers with the same length, and length > 64 will have identical toString() values.